src/frontends/ir/src/frontend.cpp

```diff
@@ -59,13 +51,25 @@ size_t get_ir_version(std::istream& model) {
     return 0;
 }
 
+size_t get_ir_version(std::istream& model) {
+    char header[512];
+
+    model.seekg(0, model.beg);
+    model.read(header, 512);
```

Comment:
Looks like a possible issue if nothing received from the istream. May suggest to check istream.gcount()